tgui/packages/tgui/interfaces/Manufacturer/components/BlueprintButton.tsx

```diff
@@ -49,17 +52,16 @@ const getProductionSatisfaction = (
   let patterns_satisfied:boolean[] = [];
   for (let i in requirement_data) {
     const target_pattern = requirement_data[i].id;
-    const target_amount = requirement_data[i].amount;
+    const target_amount = requirement_data[i].amount / 10;
     const matchingMaterial = materials_stored.find((material:ResourceData) => (
-      (material.amount >= target_amount/10) && (target_pattern === "ALL" || material.satisfies?.includes(target_pattern) || material.id === target_pattern)
+      material.amount >= target_amount && material.satisfies?.includes(target_pattern)
     ));
-    if (matchingMaterial !== undefined) {
-      material_amts_predicted[i] -= target_amount/10;
-      patterns_satisfied.push(true);
-    }
-    else {
+    if (matchingMaterial === undefined) {
       patterns_satisfied.push(false);
+      continue;
     }
+    material_amts_predicted[i] -= target_amount;
```

Comment:
No change from you required here, going to try to come back to this part of the code in the nearish future to tidy up some of the looping structures a bit.